.github/workflows/maven.yml

```diff
@@ -34,30 +34,51 @@ jobs:
-            echo "MVN_ADDITIONAL_OPTS=coveralls:report -Dlogback.configurationFile=bundle/src/test/resources/logback-test-no-logging.xml -Pcoverage,dependency-check -DrepoToken=${{ secrets.COVERALLS_TOKEN }}" >> $GITHUB_ENV
-            if [ "${{github.ref}}" = "refs/heads/master" ]; then
-              echo "MAVEN_USERNAME=${{ secrets.OSSRH_TOKEN_USER }}" >> $GITHUB_ENV
-              echo "MAVEN_PASSWORD=${{ secrets.OSSRH_TOKEN_PASSWORD }}" >> $GITHUB_ENV
-              echo "MVN_GOAL=clean deploy" >> $GITHUB_ENV
-            else
-              echo "MVN_GOAL=clean verify" >> $GITHUB_ENV
-            fi
-          else
-            echo "MVN_ADDITIONAL_OPTS=" >> $GITHUB_ENV
-            echo "MVN_GOAL=clean verify" >> $GITHUB_ENV
-          fi
-
-      - name: Build with Maven
-        run: mvn -e -B -V ${{ env.MVN_GOAL }} ${{ env.MVN_ADDITIONAL_OPTS }}
\ No newline at end of file
+#            echo "CI_NAME=Github" >> $GITHUB_ENV
+#            echo "CI_BUILD_NUMBER=${{ github.run_id }}" >> $GITHUB_ENV
+#            echo "CI_BUILD_URL=https://github.com/${{ github.repository }}/commit/${{ github.event.after }}/checks" >> $GITHUB_ENV
+#            echo "CI_BRANCH=${{ steps.refs.outputs.branch_name }}" >> $GITHUB_ENV
+#            echo "CI_PULL_REQUEST=${{ steps.refs.outputs.pr_number }}" >> $GITHUB_ENV
+#            echo "MVN_ADDITIONAL_OPTS=coveralls:report -Dlogback.configurationFile=bundle/src/test/resources/logback-test-no-logging.xml -Pcoverage,dependency-check -DrepoToken=${{ secrets.COVERALLS_TOKEN }}" >> $GITHUB_ENV
+#            if [ "${{github.ref}}" = "refs/heads/master" ]; then
+#              echo "MAVEN_USERNAME=${{ secrets.OSSRH_TOKEN_USER }}" >> $GITHUB_ENV
+#              echo "MAVEN_PASSWORD=${{ secrets.OSSRH_TOKEN_PASSWORD }}" >> $GITHUB_ENV
+#              echo "MVN_GOAL=clean deploy" >> $GITHUB_ENV
+#            else
+#              echo "MVN_GOAL=clean verify" >> $GITHUB_ENV
+#            fi
+#          else
+#            echo "MVN_ADDITIONAL_OPTS=" >> $GITHUB_ENV
+#            echo "MVN_GOAL=clean verify" >> $GITHUB_ENV
+#          fi
+
+  
+      - name: Build with Maven (Non-main builds)
+        if: ${{ !matrix.isMainBuildEnv }}
+        run: mvn -e -B -V clean verify
+   
+      - name: Build with Maven (Main build)
+        if: ${{ matrix.isMainBuildEnv }} 
+        run: mvn -e -B -V clean verify -Pcoverage,dependency-check -Dlogback.configurationFile=bundle/src/test/resources/logback-test-no-logging.xml
+        
+      - name: Upload code coverage to CodeCov (Main build)
+        if: ${{ matrix.isMainBuildEnv }}
+        run: bash <(curl -s https://codecov.io/bash)
+        
+      - name: Deploy SNAPSHOT to OSSRH     
+        if: ${{ matrix.isMainBuildEnv && github.ref == 'refs/heads/main' }}
+        env:
+          MAVEN_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.OSSRH_TOKEN }}
+        run: mvn clean deploy
```

Comment:
`mvn clean deploy` encompasses a regular Maven build.

Reply:
you thinking just do a `mvn deploy:deploy --batch-mode` ?

Reply:
No, just leave as before this PR and do the coveralls upload post deployment.

Reply:
Your master still have coveralls in it right? So that needs to come out